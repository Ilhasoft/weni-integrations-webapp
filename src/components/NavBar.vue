<template>
  <div id="topnav">
    <router-link class="link" to="discovery">{{ $t('discovery') }}</router-link>
    <router-link class="link" to="apps">{{ $t('my_apps') }}</router-link>
  </div>
</template>
<script>
  export default {
    name: 'NavBar',
  };
</script>
<style lang="scss" scoped>
  #topnav {
    font-size: $unnnic-font-size-body-lg;
    font-weight: $unnnic-font-weight-bold;
    line-height: calc($unnnic-font-size-body-lg + $unnnic-line-height-md);
    border-bottom: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
    padding-bottom: $unnnic-spacing-stack-sm;
  }

  .link {
    text-decoration: none;
    color: $unnnic-color-neutral-clean;
    margin-right: $unnnic-inline-lg;
  }

  .router-link-active {
    color: $unnnic-color-neutral-darkest;
  }
</style>
