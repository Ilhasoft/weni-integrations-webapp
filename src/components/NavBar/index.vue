<template>
  <div id="topnav">
    <router-link class="link" to="discovery">{{ $t('apps.nav.discovery') }}</router-link>
    <router-link class="link" to="my">{{ $t('apps.nav.my_apps') }}</router-link>
    <router-link class="link" to="other-apps">{{ $t('apps.nav.other_apps') }}</router-link>
  </div>
</template>
<script>
  export default {
    name: 'NavBar',
  };
</script>
<style lang="scss" scoped>
  @import './styles';
</style>
