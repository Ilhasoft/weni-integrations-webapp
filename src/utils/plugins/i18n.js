import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
