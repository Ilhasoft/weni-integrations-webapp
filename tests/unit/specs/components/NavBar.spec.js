import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import NavBar from '@/components/NavBar/index.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

describe('NavBar.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(NavBar, {
      localVue,
      i18n,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
