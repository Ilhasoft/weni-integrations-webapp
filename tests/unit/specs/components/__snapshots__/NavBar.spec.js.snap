// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue should be rendered properly 1`] = `<div id="topnav"><a class="link">some specific text</a> <a class="link">some specific text</a></div>`;
